packages/build/src/release.ts

```diff
@@ -1,93 +1,173 @@
+    console.log('mongosh: start public release.');
+
+    await releaseToDownloadCenter(zipFile, config);
+    await releaseToGithub(zipFile, githubRepo, config);
+  }
+
+  console.log('mongosh: finished release process.');
+}
+
+function safeToLogConfig(config: Config): any {
+  return {
+    version: config.version,
+    bundleId: config.bundleId,
+    input: config.input,
+    execInput: config.execInput,
+    outputDir: config.outputDir,
+    project: config.project,
+    revision: config.revision,
+    branch: config.branch,
+    isCi: config.isCi,
+    platform: config.platform,
+    repo: config.repo
+  };
+}
+
+async function compileAndZipExecutable(config: Config): Promise<ZipFile> {
   const executable = await compileExec(
     config.input,
     config.execInput,
     config.outputDir,
-    platform,
+    config.platform,
     config.analyticsConfig,
     config.segmentKey
   );
 
   // Zip the executable.
-  const artifact = await zip(executable, config.outputDir, platform, config.version);
-
-  // Sign and notarize the executable and artifact for MacOs.
-  // if (platform === Platform.MacOs) {
-  //  await publishMacOs(executable, artifact, platform, config);
-  // }
+  const artifact = await zip(
+    executable,
+    config.outputDir,
+    config.platform,
+    config.version
+  );
 
-  // Create & sign the .rpm (only on linux)
-  // Create & sign the .msi (only on win)
+  return artifact;
+}
 
-  // Upload artifact to S3 for Evergreen.
+async function releaseIntenally(artifact: ZipFile, config: Config): Promise<void> {
   await uploadArtifactToEvergreen(
-    artifact,
+    artifact.path,
     config.evgAwsKey,
     config.evgAwsSecret,
     config.project,
     config.revision
   );
+}
 
-  // Upload the artifact to downloads.10gen.com
-  // TODO: fix the release
-  //
-  // await uploadArtifactToDownloads(
-  //   artifact,
-  //   config.downloadCenterAwsKey,
-  //   config.downloadCenterAwsSecret,
-  //   config.project,
-  //   config.revision
-  // );
-
-  // // Create release and upload assets to Github. Will return true if the current
-  // // version is a new release and the release was created on Github.
-  // const octokit = new Octokit({
-  //   auth: config.githubToken,
-  //   userAgent: `mongosh ${config.version}`
-  // });
-  //
-  // const isNewRelease = await releaseToGithub(config.version, artifact, platform, octokit);
-
-  // if (isNewRelease) {
-  //   // Publish the .deb (only on linux)
-  //   // Publish the .rpm (only on linux)
-  //   // Create PR for Homebrew (only on macos)
-
-
-  //   // Create download center config and upload.
-  //   // Publish to NPM.
-  //   //
-  //   // These only need to happen once so we only run them on MacOS.
-  //   await uploadDownloadCenterConfig(
-  //     config.version,
-  //     config.downloadCenterAwsKey,
-  //     config.downloadCenterAwsSecret
-  //   );
-  // }
+async function shouldDoPublicRelease(githubRepo: GithubRepo, config: Config): Promise<boolean> {
```

Comment:
moving this function to `github-repo.ts`. It relates to that class, plus it's nicely testable in that file.